compiler/passes/src/type_checking/check_expressions.rs

```diff
@@ -594,11 +653,22 @@ impl<'a> ExpressionVisitor<'a> for TypeChecker<'a> {
                     }
 
                     // Check that the call is not to an external `inline` function.
-                    if func.variant == Variant::Inline && input.program.unwrap() != self.program_name.unwrap() {
+                    if func.variant == Variant::Inline
+                        && input.program.unwrap() != self.scope_state.program_name.unwrap()
+                    {
                         self.emit_err(TypeCheckerError::cannot_call_external_inline_function(input.span));
                     }
-
-                    let ret = self.assert_and_return_type(func.output_type, expected, input.span());
+                    // Async functions return a single future.
```

Comment:
This rule should be modified to support implicit Future typing (as in Rust).

Reply:
Should we enforce typing of program name too or just `Type`? Ex: `let f: Future<u32, u32, Future<u64>>` or `let f: Future<(u32, u32, Future<(u64), bye.aleo>), hello.aleo>`

Reply:
Should we add another parameter to `Future` so that `Future<>` and `Future` are not ambiguous? That way we can easily distinguish when the user wants to explicitly type from when they want to infer.

Reply:
`Future<>` is kind of ugly

Reply:
Is it a fair assumption to force the user to be consistent in either supplying all explicit future types or supplying none?

Reply:
We could do `Future<()>` instead

Reply:
How does this sound:
- `future` an inferred future type.
- `Future<Fn(...)->()>` for the explicit type.
It seems clunky to add locations to the type, is there a situation where we cannot infer it?

Reply:
We can always infer it, but if we wanted to give the user the ability to double check that when they write an explicit future it is the same program as the one they intended